archunit-tooling-test/build.gradle

```diff
@@ -0,0 +1,144 @@
+        }
+    }
+    main {
+        java {
+            compileClasspath += project.sourceSets.testFixtures.output
+            runtimeClasspath += project.sourceSets.testFixtures.output
+        }
+    }
+    test {
+        java {
+            compileClasspath += project.sourceSets.testFixtures.output
+            runtimeClasspath += project.sourceSets.testFixtures.output
+        }
+    }
+}
+
+jar.enabled = false
+
+task copyFixtureSources(type: Copy) {
+    into new File(project.sourceSets.testFixtures.output.resourcesDir, '@sources')
+    from(project.sourceSets.testFixtures.java) {
+        include '**/*.java'
+    }
+}
+
+processTestFixturesResources {
+    filesMatching(['**/pom.xml', '**/build.gradle']) {
+        filter ReplaceTokens, tokens: [
+                "junitDependency.version": project.property("archunit.version")
+        ]
+    }
+}
+
+dependencies {
+    implementation project(path: ':archunit-junit5-engine-api')
+    implementation project(path: ':archunit-junit5-engine')
+    implementation project(path: ':archunit-junit4')
+    implementation dependency.junit5JupiterEngine
+    implementation dependency.junit5VintageEngine
+    implementation dependency.junitPlatform
+    implementation dependency.junitPlatformLauncher
+    implementation dependency.guava
+    implementation dependency.systemLambda
+    implementation dependency.mavenInvoker
+    implementation dependency.surefireReportParser
+    implementation dependency.commonsIo
+    implementation dependency.gradleToolingApi
+    implementation dependency.surefireSharedUtils
+
+    compileOnly dependency.findBugsAnnotations
+
+    testFixturesImplementation project(path: ':archunit')
+    testFixturesImplementation project(path: ':archunit-junit5-engine-api')
+    testFixturesImplementation project(path: ':archunit-junit5-api')
+    testFixturesImplementation project(path: ':archunit-junit5-engine')
+    testFixturesImplementation project(path: ':archunit-junit4')
+    testFixturesImplementation dependency.junitPlatform
+    testFixturesImplementation dependency.junit5JupiterEngine
+    testFixturesImplementation dependency.junit5JupiterApi
+
+    testImplementation dependency.assertj
+    testImplementation dependency.junitJupiterParams
+    testImplementation dependency.junitPioneer
+}
+
+String getMavenRepoPath(Project project) {
+    def pubs = project.publishing.publications.findAll() as Set<MavenPublication>
+    def pub = pubs[0]
+    if (!pub){
+        return null
+    }
+    String groupPath = pub.groupId.replace('.', '/')
+    return "$groupPath/$pub.artifactId"
+}
+
+def resolveAgainstMavenLocal(String path) {
+    new File(new File(repositories.mavenLocal().url), path)
+}
+
+def deleteSafely(File file) {
+    if (file.exists()) {
+        println "Deleting ${file.toURI().toString()}"
+        delete file
+    } else {
+        println "Could not find ${file.toURI().toString()}"
+    }
+}
+
+task removeFromMavenLocal {
```

Comment:
This task cleans the local Maven repo to make sure we are running the Surefire and Gradle tests with an up-to-date version of ArchUnit jars